### PR TITLE
update rosetta integration

### PR DESCRIPTION
**Pull request by miiu96**
Updated elrond rosetta intergration
1. now the latest version of rosetta-sdk is used
2. fixed compile issues

Fixed proxy docker image to work again

### CODE REVIEW

File: rosetta/provider/interface.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ type ElrondProxyClient interface {
 
 	SendTransaction(tx *data.Transaction) (int, string, error)
 	ComputeTransactionHash(tx *data.Transaction) (string, error)
-	GetTransactionByHashAndSenderAddress(txHash string, sndAddr string) (*data.FullTransaction, int, error)
+	GetTransactionByHashAndSenderAddress(txHash string, sndAddr string, withResult bool) (*data.FullTransaction, int, error)
```
Comment (bogdan-rosianu) [inline rosetta/provider/interface.go:19]: withResults?

File: rosetta/services/account_service.go [hunk 2]
```diff
@@ -60,3 +60,7 @@ func (aas *accountAPIService) AccountBalance(
 
 	return response, nil
 }
+
+func (aas *accountAPIService) AccountCoins(_ context.Context, _ *types.AccountCoinsRequest) (*types.AccountCoinsResponse, *types.Error) {
```
Comment (bogdan-rosianu) [inline rosetta/services/account_service.go:64]: comment

File: rosetta/services/constants.go [hunk 3]
```diff
@@ -6,14 +6,19 @@ const (
...
 	opScResult = "SmartContractResult"
 	opInvalid  = "Invalid"
 )
+
+var (
+	opsStatusSuccess = "Success"
+	opStatusFailed   = "Failed"
+
+	// OpStatusOK is the operation status for successful operations.
+	OpStatusSuccess = &opsStatusSuccess
```
Comment (bogdan-rosianu) [inline rosetta/services/constants.go:21]: 1. wrong comment
2. why do you pass the address of the `opsStatusSuccess`? In `network_service` you then get the value from the address. I think you can use it directly, without pointers

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: rosetta/provider/interface.go (hunk 1 above)
Comment (miiu96) [reply rosetta/provider/interface.go:19]: done.

File: rosetta/services/account_service.go (hunk 2 above)
Comment (miiu96) [reply rosetta/services/account_service.go:64]: done.

File: rosetta/services/constants.go (hunk 3 above)
Comment (miiu96) [reply rosetta/services/constants.go:21]: refactored this part of code.

**Review by andreibancioiu — APPROVED** [review]

**Review by bogdan-rosianu — DISMISSED** [review]

File: Docker/Dockerfile [hunk 4]
```diff
@@ -5,13 +5,16 @@ COPY . .
 # Proxy
 WORKDIR /elrond/cmd/proxy
 RUN go build
+RUN cp /go/pkg/mod/github.com/!elrond!network/$(ls /go/pkg/mod/github.com/!elrond!network/ | grep arwen-wasm-vm | sed 's/.* //' | head -1)/wasmer/libwasmer_linux_amd64.so /lib/libwasmer_linux_amd64.so
```
[the remark this answers is not in the document]
Comment (miiu96) [reply Docker/Dockerfile:9]: done.

File: Docker/Dockerfile [hunk 5]
```diff
@@ -5,13 +5,16 @@ COPY . .
...
 
 # ===== SECOND STAGE ======
 FROM ubuntu:18.04
 COPY --from=builder /elrond/cmd/proxy /elrond/cmd/proxy
 ##Config the proxy config.toml file
 #Mount /data/config/
 
+COPY --from=builder "/lib/libwasmer_linux_amd64.so" "/lib/libwasmer_linux_amd64.so"
```
[the remark this answers is not in the document]
Comment (miiu96) [reply Docker/Dockerfile]: removed.

File: Docker/Dockerfile [hunk 6]
```diff
@@ -5,16 +5,17 @@ COPY . .
 # Proxy
 WORKDIR /elrond/cmd/proxy
 RUN go build
+# $(ls /go/pkg/mod/github.com/!elrond!network/ | grep arwen-wasm-vm | sed 's/.* //' | head -1) this command is required  to be able to exact the full name of folder that starting with 'arwen-wasm-wm'
```
Comment (bogdan-rosianu) [inline Docker/Dockerfile:8]: to exact?

File: rosetta/Dockerfile [hunk 7]
```diff
@@ -5,6 +5,7 @@ COPY . .
 # Proxy
 WORKDIR /elrond/cmd/proxy
 RUN go build
+# $(ls /go/pkg/mod/github.com/!elrond!network/ | grep arwen-wasm-vm | sed 's/.* //' | head -1) this command is required  to be able to exact the full name of folder that starting with 'arwen-wasm-wm'
```
Comment (bogdan-rosianu) [inline rosetta/Dockerfile:8]: to exact?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: Docker/Dockerfile (hunk 6 above)
Comment (miiu96) [reply Docker/Dockerfile]: done.

File: rosetta/Dockerfile (hunk 7 above)
Comment (miiu96) [reply rosetta/Dockerfile]: extract

File: Docker/Dockerfile [hunk 8]
```diff
@@ -5,7 +5,7 @@ COPY . .
 # Proxy
 WORKDIR /elrond/cmd/proxy
 RUN go build
-# $(ls /go/pkg/mod/github.com/!elrond!network/ | grep arwen-wasm-vm | sed 's/.* //' | head -1) this command is required  to be able to exact the full name of folder that starting with 'arwen-wasm-wm'
+# $(ls /go/pkg/mod/github.com/!elrond!network/ | grep arwen-wasm-vm | sed 's/.* //' | head -1) this command is required  to be able to extract the full name of folder that starting with 'arwen-wasm-wm'
```
Comment (bogdan-rosianu) [inline Docker/Dockerfile:8]: ` the full name of folder that starting with 'arwen-wasm-wm'`  =>  `the full name of the directory that starts with 'arwen-wasm-wm'`

File: rosetta/Dockerfile [hunk 9]
```diff
@@ -5,7 +5,7 @@ COPY . .
 # Proxy
 WORKDIR /elrond/cmd/proxy
 RUN go build
-# $(ls /go/pkg/mod/github.com/!elrond!network/ | grep arwen-wasm-vm | sed 's/.* //' | head -1) this command is required  to be able to exact the full name of folder that starting with 'arwen-wasm-wm'
+# $(ls /go/pkg/mod/github.com/!elrond!network/ | grep arwen-wasm-vm | sed 's/.* //' | head -1) this command is required  to be able to extract the full name of folder that starting with 'arwen-wasm-wm'
```
Comment (bogdan-rosianu) [inline rosetta/Dockerfile:8]: `the full name of folder that starting with 'arwen-wasm-wm'`  =>  `the full name of the directory that starts with 'arwen-wasm-wm'`

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**